packages/contracts-bedrock/test/L1/StandardValidator.t.sol

```diff
@@ -0,0 +1,890 @@
+import { ICrossDomainMessenger } from "interfaces/universal/ICrossDomainMessenger.sol";
+import { ISuperchainConfig } from "interfaces/L1/ISuperchainConfig.sol";
+import { IOptimismMintableERC20Factory } from "interfaces/universal/IOptimismMintableERC20Factory.sol";
+import { IL1ERC721Bridge } from "interfaces/L1/IL1ERC721Bridge.sol";
+import { IERC721Bridge } from "interfaces/universal/IERC721Bridge.sol";
+import { IPermissionedDisputeGame } from "interfaces/dispute/IPermissionedDisputeGame.sol";
+import { IProxyAdmin } from "interfaces/universal/IProxyAdmin.sol";
+import { IDelayedWETH } from "interfaces/dispute/IDelayedWETH.sol";
+import { IPreimageOracle } from "interfaces/cannon/IPreimageOracle.sol";
+import { ISemver } from "interfaces/universal/ISemver.sol";
+import { IResourceMetering } from "interfaces/L1/IResourceMetering.sol";
+import { IOptimismPortal2 } from "interfaces/L1/IOptimismPortal2.sol";
+import { IDisputeGame } from "interfaces/dispute/IDisputeGame.sol";
+import { IMIPS } from "interfaces/cannon/IMIPS.sol";
+
+contract StandardValidatorV180Testing is StandardValidatorV180 {
+    mapping(address => bytes32) internal validHashes;
+
+    constructor(
+        ISuperchainConfig _superchainConfig,
+        address _l1PAOMultisig,
+        address _mips
+    )
+        StandardValidatorV180(_superchainConfig, _l1PAOMultisig, _mips)
+    { }
+
+    function bytecodeHashEq(address _addr, bytes32 _hash) internal view override returns (bool) {
+        return validHashes[_addr] == _hash;
+    }
+
+    function setValidHash(address _addr, bytes32 _hash) external {
+        validHashes[_addr] = _hash;
+    }
+}
+
+contract StandardValidatorV180_Test is Test {
+    StandardValidatorV180Testing validator;
+    ISuperchainConfig superchainConfig;
+    address l1PAOMultisig;
+    address mips;
+
+    // Mock contracts for validation
+    IProxyAdmin proxyAdmin;
+    ISystemConfig systemConfig;
+    bytes32 absolutePrestate;
+    uint256 l2ChainID;
+
+    // Mock addresses for dependencies
+    address disputeGameFactory;
+    address permissionedDisputeGame;
+    address anchorStateRegistry;
+    address optimismPortal;
+    address l1CrossDomainMessenger;
+    address l1StandardBridge;
+    address l1ERC721Bridge;
+    address optimismMintableERC20Factory;
+    address delayedWETH;
+    address preimageOracle;
+
+    function setUp() public {
+        // Setup test addresses
+        l1PAOMultisig = makeAddr("l1PAOMultisig");
+        mips = makeAddr("mips");
+        address guardian = makeAddr("guardian");
+        superchainConfig = ISuperchainConfig(makeAddr("superchainConfig"));
+
+        // Mock superchainConfig calls needed in setup
+        vm.mockCall(address(superchainConfig), abi.encodeCall(ISuperchainConfig.guardian, ()), abi.encode(guardian));
+        vm.mockCall(address(superchainConfig), abi.encodeCall(ISuperchainConfig.paused, ()), abi.encode(false));
```

Comment:
I would suggest overwriting the slots instead, it's safer then mocking since storage slots remain accurate/consistent
```solidity
bytes32 PAUSED_SLOT = bytes32(uint256(keccak256("superchainConfig.paused")) - 1);
bytes32 GUARDIAN_SLOT = bytes32(uint256(keccak256("superchainConfig.guardian")) - 1);
vm.store(address(superchainConfig), PAUSED_SLOT, abi.encode(false));
vm.store(address(superchainConfig), GUARDIAN_SLOT, abi.encode(guardian));
```

Reply:
as per our discussion we'll use mocks